composer.json

```diff
@@ -47,7 +47,8 @@
         "jangregor/phpstan-prophecy": "^1.0",
         "phpstan/extension-installer": "^1.1",
         "vimeo/psalm": "^4.22",
-        "blackfire/php-sdk": "^1.31"
+        "blackfire/php-sdk": "^1.31",
+        "squizlabs/php_codesniffer": "^3.7"
```

Comment:
we should really stop adding these dev dependencies as they provide limited protection from breakage (only an arbitrary major version in our lock file) and would be better to have a separate pipeline for them probably. anyway, at least `phpcs` doesn't have any dependencies itself.

Reply:
Actually, we could inspire by `composer/composer` test package and create some fixtures, providing *closed* execution environment, based on their [`initTempComposer`](https://github.com/composer/composer/blob/main/tests/Composer/Test/TestCase.php#L127).

Creating a new temporary *project* for test case would spare us from keeping unnecessary dev dependencies fixed to a specific version (I guess declaring `"squizlabs/php_codesniffer": "<4"` wouldn't help either). But on the other side, such tests would be slow as it would require to install dependency during runtime.

Reply:
yeah, like I say it's fine for now, but at some point I might clean up and stub out the processes and/or execute them in separate pipelines using a similar approach to what you suggest.